packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -646,7 +646,13 @@ class TextSelectionOverlay {
     required Offset globalGesturePosition,
     required TextPosition currentTextPosition,
   }) {
-    final Offset globalRenderEditableTopLeft = renderEditable.localToGlobal(Offset.zero);
+    final RenderBox? overlay = Overlay.of(context, rootOverlay: true).context.findRenderObject() as RenderBox?;
+    final Offset overlayGesturePosition = overlay?.globalToLocal(globalGesturePosition) ?? globalGesturePosition;
+    final Offset globalRenderEditableTopLeft = renderEditable.localToGlobal(Offset.zero, ancestor: overlay);
```

Comment:
nit: get the transform and apply it to the Rect instead?

Reply:
also why is it called global if the coordinate system is just the overlay?

Reply:
1. TIL MatrixUtils.transformRect.

2. You're right, I'll change the naming to "overlay" instead of "global" and make sure all the calculations are relative to that. This will actually matter if there is a transform above the MaterialApp, so we should get it right.